benchmark/python/benchmark_hail/run/utils.py

```diff
@@ -1,15 +1,44 @@
+import contextlib
 import logging
 import os
+import re
+import signal
 import timeit
 from urllib.request import urlretrieve
 
-import numpy as np
-import re
-
 import hail as hl
+from py4j.protocol import Py4JError
+
 from .. import init_logging
 
 
+class BenchmarkTimeoutError(TimeoutError):
+    pass
+
+
+_timeout_state = False
+
+
+# https://stackoverflow.com/questions/492519/timeout-on-a-function-call/494273#494273
+@contextlib.contextmanager
+def timeout_signal(time_in_seconds):
+    global _timeout_state
+    _timeout_state = False
+
+    def handler(signum, frame):
+        global _timeout_state
+        _timeout_state = True
+        raise BenchmarkTimeoutError()
+
+    signal.signal(signal.SIGALRM, handler)
+    signal.alarm(time_in_seconds)
+
+    try:
+        yield
+    finally:
+        signal.signal(signal.SIGALRM, signal.SIG_IGN)
```

Comment:
From that stackoverflow answer, and the signal docs, it seems like `signal.alarm(0)` is the right way to cancel the timer. With `SIG_IGN`, I think the signal is still sent, and if it comes between `signal.signal(signal.SIGALRM, handler)` and `signal.alarm(time_in_seconds)` in a later benchmark, it could cause a false timeout (I think?).